### PR TITLE
WIP: Fix 1274 port `IntegerCollectionEditor`

### LINKED ISSUE #1274: Port System.Windows.Forms.Design.IntegerCollectionEditor
**Problem description:**
`IntegerCollectionEditor` from System.Windows.Forms.Design is missing

Refer to https://github.com/dotnet/winforms/issues/1115#issuecomment-504798149

### PR DESCRIPTION
**Pull request by RussKie**
Closes #1274

## Proposed changes

- Port IntegerCollectionEditor from NET Fx
- Fix xml-docs
- Fix names
- Remove unused usings
- Remove `this.`
- Add missing braces
- Use arrow functions, where appropriate

### Before

image

### After

image

## Test methodology 

- manual tests with visual inspection

### CODE REVIEW

**Comment by RussKie** [comment]
The UI appears to be different between the code available in NETFx enlistment and the official releases of .NET Fx 472/48. 
@merriemcgaw @Olina-Zhang  @dotnet/dotnet-winforms - is this expected?

The code from enlistment:
image


Offical NETFx 472/48:
image

File: src/System.Windows.Forms/tests/IntegrationTests/WinformsControlsTest/UserControls/UserControlWithCollectionEditors.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        private int[] _intArray = Enumerable.Range(1, 10).ToArray();
+        private Array _intLegacyArray = Enumerable.Range(1, 10).ToArray();
+        private List<int> _intList = Enumerable.Range(1, 10).ToList();
+
+        public UserControlWithCollectionEditors()
+        {
+            AutoScaleMode = AutoScaleMode.Font;
+        }
+
+#if NETCORE
```
Comment (danmosemsft) [inline src/System.Windows.Forms/tests/IntegrationTests/WinformsControlsTest/UserControls/UserControlWithCollectionEditors.cs]: Just curious, when is NETCORE not defined - you run these tests aginst .NET Framework too?

Comment (danmosemsft) [reply src/System.Windows.Forms/tests/IntegrationTests/WinformsControlsTest/UserControls/UserControlWithCollectionEditors.cs]: Oh I see below.

Comment (RussKie) [reply src/System.Windows.Forms/tests/IntegrationTests/WinformsControlsTest/UserControls/UserControlWithCollectionEditors.cs]: Yes, running the same code against both to allow manual/visual comparison

**Comment by Olina-Zhang** [comment]
@RussKie, in framework, when targeting .Net 4.8/4.7.2 on test machine, this IntegerCollectionEditor looks like following, not your above second one.
image
And we didn't meet this editor for Winforms any control in previous testing.

**Comment by RussKie** [comment]
This editor appears to be universally unused.
Closing.

**Pull request closed without merging**